### PR TITLE
Enable http2 health checking with go 1.16.5

### PR DESCRIPTION
**Pull request by relyt0925**
builds on: https://github.com/kubernetes-sigs/apiserver-network-proxy/pull/246

this pr will also turn on HTTP2 health checking which will auto heal the use of closed connection error.

### CODE REVIEW

**Comment by relyt0925** [comment]
holding until go 1.16 pr goes in

File: go.mod [hunk 1]
```diff
@@ -1,17 +1,17 @@
...
 	github.com/golang/mock v1.4.4
 	github.com/golang/protobuf v1.4.2
-	github.com/google/uuid v1.1.1
+	github.com/google/uuid v1.1.2
 	github.com/inconshreveable/mousetrap v1.0.0 // indirect
 	github.com/prometheus/client_golang v1.7.1
 	github.com/spf13/cobra v0.0.3
 	github.com/spf13/pflag v1.0.5
 	golang.org/x/net v0.0.0-20201110031124-69a78807bb2b
-	google.golang.org/grpc v1.27.0
```
Comment (cheftako) [inline go.mod:14]: Not against upgrading grpc (I tested this myself a few weeks ago) but I would rather we separated this upgrade from the golang upgrade.

Comment (relyt0925) [reply go.mod:14]: great point I agree

Comment (relyt0925) [reply go.mod]: done

**Comment by relyt0925** [comment]
cc @cheftako this should be good based off initial comments (or ready for another round of comments)

File: cmd/client/main.go [hunk 2]
```diff
@@ -259,6 +261,10 @@ func (c *Client) run(o *GrpcProxyClientOptions) error {
 			transport := &http.Transport{
 				DialContext: dialer,
 			}
+			err = configureHTTP2Transport(transport)
```
Comment (cheftako) [inline cmd/client/main.go:264]: Can we wrap this in a CLI flag to control whether we have it turned on or not?
Otherwise looks good, thank you

Comment (relyt0925) [reply cmd/client/main.go:264]: can do!

Comment (relyt0925) [reply cmd/client/main.go]: done

**Pull request merged**